src/config.cc

```diff
@@ -0,0 +1,47 @@
+#include "config.h"
+
+namespace {
+struct config_singleton_t {
+protected:
+  boost::property_tree::ptree config_;
+
+  config_singleton_t() = delete;
+
+  config_singleton_t(const std::string& config_file_or_inline) {
+    if (config_file_or_inline.empty()) {
+      throw valhalla::ConfigUninitializedException();
```

Comment:
while you're at it, can we please remove this custom error from the codebase and make it a good ol' `std::runtime_error` with the same message?

Reply:
ok, changed it